JSTests/stress/bound-function-tail-call.js

```diff
@@ -0,0 +1,43 @@
+//@ requireOptions("--useConcurrentJIT=0", "--dfgAllowlist=test4", "--useFTLJIT=0")
+function shouldBe(actual, expected) {
+    if (actual !== expected)
+        throw new Error('bad value: ' + actual);
+}
+
+(function entry() {
+    "use strict";
+    var count = 0;
+    var flag = false;
+    flag = true;
+    flag = false;
+    function test() {
+        ++count;
+        return 42;
+    }
+    noInline(test);
+
+    function test2()
+    {
+        ++count;
```

Comment:
It would be nice to know why this OSR exits for perf, we probably shouldn't. Maybe we could also add an explicit $vm.osrExit to make the test more robust.